internal/errorlint/errorlint.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: See package comment for this one file package.
+
+// Package errorlint contains the necessary logic for the error/log/trace linter.
+package errorlint
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/getoutreach/lintroller/internal/common"
+	"github.com/getoutreach/lintroller/internal/reporter"
+	"golang.org/x/tools/go/analysis"
+)
+
+// name defines the name for the errorlint linter.
+const name = "errorlint"
+
+// doc defines the help text for the error linter.
+const doc = `Ensures that each static message in error/trace/log/ is lowercase.
```

Comment:
Another idiom of golang error strings is that they shouldn't end in punctuation (because they're often wrapped). I haven't made it to the rest of your code yet, so maybe you already check against this and if that's the case you should augment the doc text, but if you haven't this might be a good thing to add to this linter.

Reply:
I already added it that it should not end in ". : ! and \n " is stringFormatted function is checking that.

Reply:
Yes I did end up seeing that, which is awesome. You should make sure _everything_ the linter does is noted in this doc text though

Reply:
sure, ill add descriptive comment

Reply:
added